From 6ebe4bda99fd5af41375a2ff582c28850f3cf436 Mon Sep 17 00:00:00 2001
From: Alex Naidis <alex.naidis@linux.com>
Date: Thu, 13 Oct 2016 19:25:52 +0200
Subject: [PATCH 6/6] skia: update support for advanced compiler features

* set flags to be passed to linker
* add FDO support to the skia static lib

Change-Id: I38a2fbfa1648d34c4890e73c4aebe965d5a64d71
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
---
 Android.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Android.mk b/Android.mk
index b073351..2a6f66a 100644
--- a/Android.mk
+++ b/Android.mk
@@ -44,6 +44,7 @@ LOCAL_PATH:= $(call my-dir)
 
 include $(CLEAR_VARS)
 LOCAL_SDCLANG_LTO := true
+LOCAL_SDCLANG_LTO_LDFLAGS := -O3 -fPIC -funsafe-math-optimizations
 LOCAL_FDO_SUPPORT := true
 # This should be the last -Oxxx specified in LOCAL_CFLAGS
 LOCAL_CFLAGS += -O3
@@ -830,6 +831,8 @@ LOCAL_WHOLE_STATIC_LIBRARIES := libskia_static
 LOCAL_SHARED_LIBRARIES := \
         libcutils
 
+LOCAL_FDO_SUPPORT := true
+
 #ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
 #LOCAL_WHOLE_STATIC_LIBRARIES += libqc-skia
 #endif
-- 
2.10.0

